### PR TITLE
uintptr, unsafe_pointer: implement

### LINKED ISSUE #88: Shall we add support of uintptr and Pointer?
I'll be glad to work on it if it is good to go.

```go
func LoadUintptr(addr *uintptr) (val uintptr)
func StoreUintptr(addr *uintptr, val uintptr)
func AddUintptr(addr *uintptr, delta uintptr) (new uintptr)
func SwapUintptr(addr *uintptr, new uintptr) (old uintptr)
func CompareAndSwapUintptr(addr *uintptr, old, new uintptr) (swapped bool)

func LoadPointer(addr *unsafe.Pointer) (val unsafe.Pointer)
func StorePointer(addr *unsafe.Pointer, val unsafe.Pointer)
func SwapPointer(addr *unsafe.Pointer, new unsafe.Pointer) (old unsafe.Pointer)
func CompareAndSwapPointer(addr *unsafe.Pointer, old, new unsafe.Pointer) (swapped bool)
```

### PR DESCRIPTION
**Pull request by tisonkun**
closed #88

Before opening your pull request, please make sure that you've:

- [x] updated the changelog if the change is user-facing;
- [x] [signed Uber's Contributor License Agreement](https://docs.google.com/a/uber.com/forms/d/1pAwS_-dA1KhPlfxzYLBqK6rsSWwRwH95OCCZrcsY5rk/viewform);
- [x] added tests to cover your changes;
- [x] run the test suite locally (`make test`); and finally,
- [x] run the linters locally (`make lint`).

### CODE REVIEW

**Comment by tisonkun** [comment]
It seems we change the CLA process to https://cla-assistant.io/uber-go/atomic?pullRequest=90, instead of [signed Uber's Contributor License Agreement](https://docs.google.com/a/uber.com/forms/d/1pAwS_-dA1KhPlfxzYLBqK6rsSWwRwH95OCCZrcsY5rk/viewform). I'm a bit down on doing the extra work.

**Comment by tisonkun** [comment]
> It seems we change the CLA process to https://cla-assistant.io/uber-go/atomic?pullRequest=90, instead of [signed Uber's Contributor License Agreement](https://docs.google.com/a/uber.com/forms/d/1pAwS_-dA1KhPlfxzYLBqK6rsSWwRwH95OCCZrcsY5rk/viewform). I'm a bit down on doing the extra work.

A separated PR https://github.com/uber-go/atomic/pull/91

**Review by abhinav — APPROVED** [review]
Thanks for the contribution. This looks great!

It's too bad that UnsafePointer cannot be auto-generated without significant changes to the code generator, so I'm okay deferring that for now.

File: uintptr.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
+// @generated Code generated by gen-atomicint.
+
+// Copyright (c) 2020 Uber Technologies, Inc.
```
[the remark this answers is not in the document]
Comment (tisonkun) [reply uintptr.go:3]: Yep. I did think of update it later this PR but it is good you fix it :-)

[the remark this answers is not in the document]
Comment (tisonkun) [reply uintptr.go:3]:
```suggestion
// Copyright (c) 2021 Uber Technologies, Inc.
```

File: unsafe_pointer_test.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+package atomic
+
+import (
+	"github.com/stretchr/testify/require"
+	"testing"
+	"unsafe"
+)
+
+func TestUnsafePointer(t *testing.T) {
```
Comment (prashantv) [inline unsafe_pointer_test.go:29]: Can we add a couple of test cases for `nil` as well:
 * If we do `var val UnsafePointer` then `Load() == unsafe.Pointer(nil)`
 * store/load work fine with `nil`

**Review by prashantv — APPROVED** [review]
LGTM with minor minor test case additions.

File: uintptr.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (abhinav) [reply uintptr.go]: #92 merged. Would you mind merging that in and regenerating? Thanks!

File: unsafe_pointer_test.go (hunk 2 above)
Comment (abhinav) [reply unsafe_pointer_test.go:29]: turned into a table test to verify all cases with nil and non-nil versions

**Review by prashantv — APPROVED** [review]

**Comment by tisonkun** [comment]
Thanks for your helps and merging! I just woke up :P
